--- irqbalance/irqbalance.h
+++ irqbalance/irqbalance.h
@@ -163,7 +163,7 @@
 #endif /* HAVE_LIBSYSTEMD */
 
 #define SOCKET_PATH "irqbalance"
-#define SOCKET_TMPFS "/run/irqbalance"
+#define SOCKET_TMPFS "/var/run/irqbalance"
 
 extern int process_one_line(char *path, void (*cb)(char *line, void *data), void *data);
 extern void get_mask_from_bitmap(char *line, void *mask);
--- irqbalance/ui/irqbalance-ui.h
+++ irqbalance/ui/irqbalance-ui.h
@@ -8,7 +8,7 @@
 #include <glib-unix.h>
 
 #define SOCKET_PATH "irqbalance"
-#define SOCKET_TMPFS "/run/irqbalance"
+#define SOCKET_TMPFS "/var/run/irqbalance"
 
 #define STATS "stats"
 #define SET_SLEEP "settings sleep "
