From f4c55efa0f46df97bef42d530e9c0953697b596a Mon Sep 17 00:00:00 2001
From: lancethepants <lancethepants@gmail.com>
Date: Thu, 28 Aug 2025 10:45:43 -0600
Subject: [PATCH] xl2tpd: fix compilation with gcc15

---
 xl2tpd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/xl2tpd.c b/xl2tpd.c
index 791d5a4..771f28a 100644
--- a/xl2tpd.c
+++ b/xl2tpd.c
@@ -74,7 +74,7 @@ static int control_handle_lac_hangup(FILE* resf, char* bufp);
 static int control_handle_lac_disconnect(FILE* resf, char* bufp);
 static int control_handle_lac_add_modify(FILE* resf, char* bufp);
 static int control_handle_lac_remove(FILE* resf, char* bufp);
-static int control_handle_lac_status();
+static int control_handle_lac_status(FILE *fp, char *arg);
 static int control_handle_lns_remove(FILE* resf, char* bufp);
 
 static struct control_requests_handler control_handlers[] = {
@@ -1549,7 +1549,7 @@ static int control_handle_lac_remove(FILE* resf, char* bufp){
     return 1;
 }
 
-static int control_handle_lac_status(){
+static int control_handle_lac_status(FILE *fp, char *arg){
     show_status ();
     return 1;
 }
-- 
2.39.5

