diff -Naur ffmpeg-6.1.2.orig/libavdevice/v4l2.c ffmpeg-6.1.2/libavdevice/v4l2.c
--- ffmpeg-6.1.2.orig/libavdevice/v4l2.c	2024-08-12 07:34:27.048777565 -0600
+++ ffmpeg-6.1.2/libavdevice/v4l2.c	2024-08-12 07:34:46.804764558 -0600
@@ -107,11 +107,7 @@
     int (*open_f)(const char *file, int oflag, ...);
     int (*close_f)(int fd);
     int (*dup_f)(int fd);
-#if defined(__sun) || defined(__BIONIC__) || defined(__musl__) /* POSIX-like */
     int (*ioctl_f)(int fd, int request, ...);
-#else
-    int (*ioctl_f)(int fd, unsigned long int request, ...);
-#endif
     ssize_t (*read_f)(int fd, void *buffer, size_t n);
     void *(*mmap_f)(void *start, size_t length, int prot, int flags, int fd, int64_t offset);
     int (*munmap_f)(void *_start, size_t length);
