From 433ae56315c18d418c493ce88c39465979bb5222 Mon Sep 17 00:00:00 2001
From: lancethepants <lancethepants@gmail.com>
Date: Wed, 13 Nov 2024 14:07:30 -0700
Subject: [PATCH] Dropbear 2024.86

---
 localoptions.h   | 24 +++++++++++++++++++++++-
 src/sysoptions.h |  2 ++
 2 files changed, 25 insertions(+), 1 deletion(-)

diff --git a/localoptions.h b/localoptions.h
index 890ab7e..94c6e1b 100644
--- a/localoptions.h
+++ b/localoptions.h
@@ -3,4 +3,26 @@
 #endif
 #define DO_HOST_LOOKUP 1
 #define DROPBEAR_SMALL_CODE 0
-#define DEFAULT_PATH "/bin:/sbin:/usr/bin:/usr/sbin"
+
+/* Override SSH 2.0 ident */
+#define LOCAL_IDENT "SSH-2.0-dropbear"
+
+#define DROPBEAR_X11FWD 0
+#define INETD_MODE 0
+
+/* encryption/hostkey/public-key algo (disable 3DES, DSS) */
+#define DROPBEAR_3DES 0
+#define DROPBEAR_DSS 0
+
+#define XAUTH_COMMAND "/opt/X11R6/X11/xauth -q"
+#define SFTPSERVER_PATH "/opt/libexec/sftp-server"
+
+#define DROPBEAR_ENABLE_CBC_MODE 0
+#define DROPBEAR_SHA1_96_HMAC 0
+#define DEFAULT_PATH "/bin:/usr/bin:/sbin:/usr/sbin:/opt/bin:/opt/sbin:/opt/usr/bin:/opt/usr/sbin"
+#define DEFAULT_ROOT_PATH "/bin:/usr/bin:/sbin:/usr/sbin:/opt/bin:/opt/sbin:/opt/usr/bin:/opt/usr/sbin"
+
+#define MAX_UNAUTH_PER_IP 2  /* default 5 */
+#define MAX_UNAUTH_CLIENTS 5 /* default 30 */
+#define MAX_AUTH_TRIES 2     /* default 10 */
+#define AUTH_TIMEOUT 60      /* default 300 */
diff --git a/src/sysoptions.h b/src/sysoptions.h
index 9b325d5..5822ec0 100644
--- a/src/sysoptions.h
+++ b/src/sysoptions.h
@@ -11,7 +11,9 @@
 #ifndef IDENT_VERSION_PART
 #define IDENT_VERSION_PART "_" DROPBEAR_VERSION
 #endif
+#ifndef LOCAL_IDENT
 #define LOCAL_IDENT "SSH-2.0-dropbear" IDENT_VERSION_PART
+#endif
 #define PROGNAME "dropbear"
 
 #ifndef DROPBEAR_CLIENT
-- 
2.39.5

