From 0732d7d8b3497b8f4c60e4d048cdd0ddf3eb2825 Mon Sep 17 00:00:00 2001
From: lancethepants <lancethepants@gmail.com>
Date: Wed, 20 Nov 2024 13:44:22 -0700
Subject: [PATCH] fix include for tomato64

---
 src/discovery.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/discovery.c b/src/discovery.c
index 42dcae9..6b0e617 100644
--- a/src/discovery.c
+++ b/src/discovery.c
@@ -38,9 +38,9 @@
 #define HAVE_STDARG_H 1
 #define HAVE_STDBOOL_H 1
 #define HAVE_STDDEF_H 1
-#include "../../pppd/pppd/pppd.h"
-#include "../../pppd/pppd/fsm.h"
-#include "../../pppd/pppd/lcp.h"
+#include "pppd/pppd.h"
+#include "pppd/fsm.h"
+#include "pppd/lcp.h"
 
 #ifdef PPPD_VERSION
 /* New-style pppd API */
-- 
2.39.5

