From 8e69987fbbf726e6d9f081fe32adc6e0febe6739 Mon Sep 17 00:00:00 2001
From: lancethepants <lancethepants@gmail.com>
Date: Tue, 4 Jun 2024 14:39:28 -0600
Subject: [PATCH] rdma/rdma.h: add include to fix with musl 1.2.5

---
 rdma/rdma.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rdma/rdma.h b/rdma/rdma.h
index df1852db..005d5dbf 100644
--- a/rdma/rdma.h
+++ b/rdma/rdma.h
@@ -9,6 +9,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <errno.h>
+#include <libgen.h>
 #include <getopt.h>
 #include <netinet/in.h>
 #include <libmnl/libmnl.h>
-- 
2.39.2

