--- miniupnpd/miniupnpd.c
+++ miniupnpd/miniupnpd.c
@@ -1145,18 +1145,18 @@
 			syslog(LOG_INFO, "STUN: ext interface %s has now public IP address %s and firewall does not block incoming connections set by miniupnpd", ext_if_name, if_addr_str);
 		syslog(LOG_INFO, "Port forwarding is now enabled");
 	} else if ((init || !disable_port_forwarding) && restrictive_nat) {
-		if (addr_is_reserved(&if_addr)) {
-			syslog(LOG_WARNING, "STUN: ext interface %s with private IP address %s is now possibly behind restrictive or symmetric NAT with public IP address %s which does not support port forwarding", ext_if_name, if_addr_str, ext_addr_str);
-			syslog(LOG_WARNING, "NAT on upstream router blocks incoming connections set by miniupnpd");
-			syslog(LOG_WARNING, "Turn off NAT on upstream router or change it to full-cone NAT 1:1 type");
-		} else {
-			syslog(LOG_WARNING, "STUN: ext interface %s has now public IP address %s but firewall filters incoming connections set by miniunnpd", ext_if_name, if_addr_str);
-			syslog(LOG_WARNING, "Check configuration of firewall on local machine and also on upstream router");
-		}
-		if (!GETFLAG(ALLOWFILTEREDSTUNMASK)) {
-			syslog(LOG_WARNING, "Port forwarding is now disabled");
-			syslog(LOG_WARNING, "Set ext_perform_stun=allow-filtered if you still want to use port forwarding in current situation");
-		}
+//		if (addr_is_reserved(&if_addr)) {
+//			syslog(LOG_WARNING, "STUN: ext interface %s with private IP address %s is now possibly behind restrictive or symmetric NAT with public IP address %s which does not support port forwarding", ext_if_name, if_addr_str, ext_addr_str);
+//			syslog(LOG_WARNING, "NAT on upstream router blocks incoming connections set by miniupnpd");
+//			syslog(LOG_WARNING, "Turn off NAT on upstream router or change it to full-cone NAT 1:1 type");
+//		} else {
+//			syslog(LOG_WARNING, "STUN: ext interface %s has now public IP address %s but firewall filters incoming connections set by miniunnpd", ext_if_name, if_addr_str);
+//			syslog(LOG_WARNING, "Check configuration of firewall on local machine and also on upstream router");
+//		}
+//		if (!GETFLAG(ALLOWFILTEREDSTUNMASK)) {
+//			syslog(LOG_WARNING, "Port forwarding is now disabled");
+//			syslog(LOG_WARNING, "Set ext_perform_stun=allow-filtered if you still want to use port forwarding in current situation");
+//		}
 	} else {
 		syslog(LOG_INFO, "STUN: ... done");
 	}
