--- miniupnpd/miniupnpd.c
+++ miniupnpd/miniupnpd.c
@@ -1014,11 +1014,13 @@ 
 				INIT_PRINT_ERR("Error parsing address : %s\n", lan_addr->ext_ip_str);
 				return -1;
 			}
+#if 0
 			if(addr_is_reserved(&lan_addr->ext_ip_addr)) {
 				/* error */
 				INIT_PRINT_ERR("Error: option ext_ip address contains reserved / private address : %s\n", lan_addr->ext_ip_str);
 				return -1;
 			}
+#endif
 		}
 	}
 #else
@@ -1851,10 +1853,12 @@ 
 			INIT_PRINT_ERR("Error: option ext_ip contains invalid address %s\n", use_ext_ip_addr);
 			return 1;
 		}
+#if 0
 		if (addr_is_reserved(&addr)) {
 			INIT_PRINT_ERR("Error: option ext_ip contains reserved / private address %s, not public routable\n", use_ext_ip_addr);
 			return 1;
 		}
+#endif
 	}
 
 #ifndef NO_BACKGROUND_NO_PIDFILE
--- miniupnpd/natpmp.c
+++ miniupnpd/natpmp.c
@@ -109,9 +109,11 @@ 
 			syslog(LOG_ERR, "Failed to get IP for interface %s", ext_if_name);
 			resp[3] = 3;	/* Network Failure (e.g. NAT box itself
 			                 * has not obtained a DHCP lease) */
+#if 0
 		} else if (addr_is_reserved(&addr)) {
 			resp[3] = 3;	/* Network Failure, box has not obtained
 			                   public IP address */
+#endif
 		} else {
 			inet_pton(AF_INET, tmp, resp+8); /* ok */
 		}
--- miniupnpd/upnpdescgen.c
+++ miniupnpd/upnpdescgen.c
@@ -1316,7 +1316,7 @@ 
 				else {
 					struct in_addr addr;
 					char ext_ip_addr[INET_ADDRSTRLEN];
-					if(getifaddr(ext_if_name, ext_ip_addr, INET_ADDRSTRLEN, &addr, NULL) < 0 || addr_is_reserved(&addr)) {
+					if(getifaddr(ext_if_name, ext_ip_addr, INET_ADDRSTRLEN, &addr, NULL) < 0) {
 						str = strcat_str(str, len, &tmplen, "0.0.0.0");
 					} else {
 						str = strcat_str(str, len, &tmplen, ext_ip_addr);
--- miniupnpd/upnpsoap.c
+++ miniupnpd/upnpsoap.c
@@ -346,9 +346,11 @@ 
 			syslog(LOG_ERR, "Failed to get ip address for interface %s",
 				ext_if_name);
 			ext_ip_addr[0] = '\0';
+#if 0
 		} else if (addr_is_reserved(&addr)) {
 			syslog(LOG_NOTICE, "private/reserved address %s is not suitable for external IP", ext_ip_addr);
 			ext_ip_addr[0] = '\0';
+#endif
 		}
 	}
 #else
