diff -Naur buildroot-2023.05.orig/package/libcurl/libcurl.mk buildroot-2023.05/package/libcurl/libcurl.mk
--- buildroot-2023.05.orig/package/libcurl/libcurl.mk	2023-06-27 23:12:43.710896565 -0600
+++ buildroot-2023.05/package/libcurl/libcurl.mk	2023-06-27 23:13:30.826842273 -0600
@@ -64,7 +64,8 @@
 ifeq ($(BR2_PACKAGE_LIBCURL_OPENSSL),y)
 LIBCURL_DEPENDENCIES += openssl
 LIBCURL_CONF_OPTS += --with-openssl=$(STAGING_DIR)/usr \
-	--with-ca-path=/etc/ssl/certs
+	--with-ca-path=/etc/ssl/certs \
+	--with-ca-bundle=/etc/ssl/certs/ca-certificates.crt
 else
 LIBCURL_CONF_OPTS += --without-openssl
 endif
