From 54f98cd322aaf3f01a973d53afcb32d99412defc Mon Sep 17 00:00:00 2001
From: Lance Fredrickson <lancethepants@gmail.com>
Date: Tue, 3 Sep 2024 08:16:08 -0600
Subject: [PATCH] libcurl: Add ca bundle

---
 package/libcurl/libcurl.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/package/libcurl/libcurl.mk b/package/libcurl/libcurl.mk
index 71fb6152c7..158d52dc53 100644
--- a/package/libcurl/libcurl.mk
+++ b/package/libcurl/libcurl.mk
@@ -67,7 +67,8 @@ endif
 ifeq ($(BR2_PACKAGE_LIBCURL_OPENSSL),y)
 LIBCURL_DEPENDENCIES += openssl
 LIBCURL_CONF_OPTS += --with-openssl=$(STAGING_DIR)/usr \
-	--with-ca-path=/etc/ssl/certs
+	--with-ca-path=/etc/ssl/certs \
+	--with-ca-bundle=/etc/ssl/certs/ca-certificates.crt
 else
 LIBCURL_CONF_OPTS += --without-openssl
 endif
-- 
2.39.5

