diff -Naur buildroot-2024.02-rc2.orig/package/ccache/ccache.mk buildroot-2024.02-rc2/package/ccache/ccache.mk
--- buildroot-2024.02-rc2.orig/package/ccache/ccache.mk	2024-03-04 15:00:10.734053171 -0700
+++ buildroot-2024.02-rc2/package/ccache/ccache.mk	2024-03-04 15:00:27.654082003 -0700
@@ -17,7 +17,8 @@
 	-UCMAKE_CXX_COMPILER_LAUNCHER \
 	-DZSTD_FROM_INTERNET=OFF \
 	-DHIREDIS_FROM_INTERNET=OFF \
-	-DENABLE_TESTING=OFF
+	-DENABLE_TESTING=OFF \
+	-DHAVE_C_AVX512=no
 
 # Patch host-ccache as follows:
 #  - Use BR_CACHE_DIR instead of CCACHE_DIR, because CCACHE_DIR
