From 858d7e3f258ae22eec3fb6683ff8f89cefd59425 Mon Sep 17 00:00:00 2001
From: lancethepants <lancethepants@gmail.com>
Date: Fri, 2 May 2025 13:56:27 -0600
Subject: [PATCH] hotplug2: run packet-steering on net events

---
 examples/hotplug2.rules-2.6kernel | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/examples/hotplug2.rules-2.6kernel b/examples/hotplug2.rules-2.6kernel
index e4fed2c..ee1aa58 100644
--- a/examples/hotplug2.rules-2.6kernel
+++ b/examples/hotplug2.rules-2.6kernel
@@ -10,6 +10,10 @@ SUBSYSTEM == ieee80211 {
 	exec /usr/bin/wlconfig ;
 }
 
+SUBSYSTEM == net {
+	exec /usr/bin/packet-steering ;
+}
+
 SUBSYSTEM == usb, DEVICENAME ~~ ^(lp[0-9]+)$, DEVPATH is set, ACTION == add {
 	exec mkdir -p -m 0777 /dev/usb ;
 	symlink /dev/%DEVICENAME% /dev/usb/%DEVICENAME%
-- 
2.39.5

