--- minidlna/upnphttp.c
+++ minidlna/upnphttp.c
@@ -913,7 +913,7 @@
 		h->req_buflen = endbuf - h->req_buf;
 		h->state = 100;
 	}
-
+#if 0 /* DNS rebind protection breaks access through hostname */
 	DPRINTF(E_DEBUG, L_HTTP, "HTTP REQUEST: %.*s\n", h->req_buflen, h->req_buf);
 	if(h->req_Host && h->req_HostLen > 0) {
 		const char *port = memchr(h->req_Host, ':', h->req_HostLen);
@@ -940,6 +940,7 @@
 			return;
 		}
 	}
+#endif
 	if(strcmp("POST", HttpCommand) == 0)
 	{
 		h->req_command = EPost;
