--- busybox/include/libbb.h
+++ busybox/include/libbb.h
@@ -2282,7 +2282,7 @@
 /* util-linux manpage says /sbin:/bin:/usr/sbin:/usr/bin,
  * but I want to save a few bytes here:
  */
-#define bb_default_path      (bb_PATH_root_path + sizeof("PATH=/sbin:/usr/sbin"))
+#define bb_default_path      (bb_PATH_root_path + sizeof("PATH=/sbin:/usr/sbin:/opt/sbin"))
 
 extern const int const_int_0;
 //extern const int const_int_1;
