--- busybox/scripts/kconfig/lxdialog/colors.h
+++ busybox/scripts/kconfig/lxdialog/colors.h
@@ -117,11 +117,11 @@
 #define TAG_SELECTED_BG              COLOR_BLUE
 #define TAG_SELECTED_HL              TRUE
 
-#define TAG_KEY_FG                   COLOR_YELLOW
+#define TAG_KEY_FG                   COLOR_RED
 #define TAG_KEY_BG                   COLOR_WHITE
 #define TAG_KEY_HL                   TRUE
 
-#define TAG_KEY_SELECTED_FG          COLOR_YELLOW
+#define TAG_KEY_SELECTED_FG          COLOR_GREEN
 #define TAG_KEY_SELECTED_BG          COLOR_BLUE
 #define TAG_KEY_SELECTED_HL          TRUE
 
