--- pptpd/defaults.h	2014-02-18 12:16:42.000000000 +0200
+++ pptpd/defaults.h	2018-04-22 17:34:39.000000000 +0200
@@ -27,7 +27,7 @@
 
 /* Default configuration values, mostly configurable */
 
-#define CONNECTIONS_DEFAULT             100
+#define CONNECTIONS_DEFAULT             6
 #define DEFAULT_LOCAL_IP_LIST           "192.168.0.1-100"
 #define DEFAULT_REMOTE_IP_LIST          "192.168.1.1-100"
 
@@ -50,9 +50,9 @@
 
 /* Location of binaries */
 
-#define PPTP_CTRL_BIN                   SBINDIR "/pptpctrl"
-#define PPTPD_BIN                       SBINDIR "/pptpd"
-#define BCRELAY_BIN                     SBINDIR "/bcrelay"
+#define PPTP_CTRL_BIN                   "/usr/sbin/pptpctrl"
+#define PPTPD_BIN                       "/usr/sbin/pptpd"
+#define BCRELAY_BIN                     "/usr/sbin/bcrelay"
 
 /* Parameters permitted in the config file */
 
--- pptpd/plugins/pptpd-logwtmp.c	2013-02-18 12:16:42.000000000 +0200
+++ pptpd/plugins/pptpd-logwtmp.c	2018-04-22 17:34:57.000000000 +0200
@@ -12,8 +12,8 @@
 #include <unistd.h>
 #include <utmp.h>
 #include <string.h>
-#include <pppd/pppd.h>
+#include "pppd/pppd.h"
 
 char pppd_version[] = VERSION;
 
 static char pptpd_original_ip[PATH_MAX+1];
