--- avahi/avahi-daemon/main.c
+++ avahi/avahi-daemon/main.c
@@ -115,6 +115,8 @@
     unsigned n_entries_per_entry_group_max;
 #endif
     int drop_root;
+    char *user;
+    char *group;
     int set_rlimits;
 #ifdef ENABLE_CHROOT
     int use_chroot;
@@ -764,6 +766,12 @@
 
                     c->n_entries_per_entry_group_max = k;
 #endif
+                } else if (strcasecmp(p->key, "user") == 0) {
+                    avahi_free(c->user);
+                    c->user = avahi_strdup(p->value);
+                } else if (strcasecmp(p->key, "group") == 0) {
+                    avahi_free(c->group);
+                    c->group = avahi_strdup(p->value);
                 } else {
                     avahi_log_error("Invalid configuration key \"%s\" in group \"%s\"\n", p->key, g->name);
                     goto finish;
@@ -1341,21 +1349,25 @@
 static int drop_root(void) {
     struct passwd *pw;
     struct group * gr;
+    const char *user, *group;
     int r;
 
-    if (!(pw = getpwnam(AVAHI_USER))) {
-        avahi_log_error( "Failed to find user '"AVAHI_USER"'.");
+    user = config.user ? config.user : AVAHI_USER;
+    group = config.group ? config.group : AVAHI_GROUP;
+
+    if (!(pw = getpwnam(user))) {
+        avahi_log_error( "Failed to find user '%s'.", user);
         return -1;
     }
 
-    if (!(gr = getgrnam(AVAHI_GROUP))) {
-        avahi_log_error( "Failed to find group '"AVAHI_GROUP"'.");
+    if (!(gr = getgrnam(group))) {
+        avahi_log_error( "Failed to find group '%s'.", group);
         return -1;
     }
 
-    avahi_log_info("Found user '"AVAHI_USER"' (UID %lu) and group '"AVAHI_GROUP"' (GID %lu).", (unsigned long) pw->pw_uid, (unsigned long) gr->gr_gid);
+    avahi_log_info("Found user '%s' (UID %lu) and group '%s' (GID %lu).", user, (unsigned long) pw->pw_uid, group, (unsigned long) gr->gr_gid);
 
-    if (initgroups(AVAHI_USER, gr->gr_gid) != 0) {
+    if (initgroups(user, gr->gr_gid) != 0) {
         avahi_log_error("Failed to change group list: %s", strerror(errno));
         return -1;
     }
@@ -1412,14 +1424,18 @@
     struct passwd *pw;
     struct group * gr;
     struct stat st;
+    const char *user, *group;
 
-    if (!(pw = getpwnam(AVAHI_USER))) {
-        avahi_log_error( "Failed to find user '"AVAHI_USER"'.");
+    user = config.user ? config.user : AVAHI_USER;
+    group = config.group ? config.group : AVAHI_GROUP;
+
+    if (!(pw = getpwnam(user))) {
+        avahi_log_error( "Failed to find user '%s'.", user);
         goto fail;
     }
 
-    if (!(gr = getgrnam(AVAHI_GROUP))) {
-        avahi_log_error( "Failed to find group '"AVAHI_GROUP"'.");
+    if (!(gr = getgrnam(group))) {
+        avahi_log_error( "Failed to find group '%s'.", group);
         goto fail;
     }
 
@@ -1527,6 +1543,8 @@
 #endif
 
     config.drop_root = 1;
+    config.user = NULL;
+    config.group = NULL;
     config.set_rlimits = 1;
 #ifdef ENABLE_CHROOT
     config.use_chroot = 1;
@@ -1697,6 +1715,8 @@
 
     avahi_server_config_free(&config.server_config);
     avahi_free(config.config_file);
+    avahi_free(config.user);
+    avahi_free(config.group);
     avahi_strfreev(config.publish_dns_servers);
     avahi_strfreev(resolv_conf_name_servers);
     avahi_strfreev(resolv_conf_search_domains);
