--- pppd/pppd/fsm.c
+++ pppd/pppd/fsm.c
@@ -71,6 +71,8 @@
 
 int peer_mru[NUM_PPP];
 
+/* add to wanstatus.log */
+void saveWANStatus(char *currentstatus, int statusindex);
 
 /*
  * fsm_init - Initialize fsm.
@@ -298,6 +300,10 @@
     case ACKRCVD:
     case ACKSENT:
 	if (f->retransmits <= 0) {
+	    /* add to wanstatus.log */
+	    int statusindex = 0; 
+	    saveWANStatus("No response from ISP.", statusindex);
+
 	    warn("%s: timeout sending Config-Requests\n", PROTO_NAME(f));
 	    f->state = STOPPED;
 	    if( (f->flags & OPT_PASSIVE) == 0 && f->callbacks->finished )
--- pppd/pppd/lcp.c
+++ pppd/pppd/lcp.c
@@ -72,6 +72,9 @@
 
 static void lcp_delayed_up(void *);
 
+/* add to wanstatus.log */
+void saveWANStatus(char *currentstatus, int statusindex);
+
 /*
  * These definitions relate to the measurement and logging of round-trip
  * time (RTT) of LCP echo-requests implemented in lcp_rtt_update_buffer().
@@ -408,6 +411,7 @@
 lcp_close(int unit, char *reason)
 {
     fsm *f = &lcp_fsm[unit];
+    int statusindex = 0; /* add to wanstatus.log */
     int oldstate;
 
     if (!in_phase(PHASE_DEAD) && !in_phase(PHASE_MASTER))
@@ -420,6 +424,16 @@
     oldstate = f->state;
 
     fsm_close(f, reason);
+    /* add to wanstatus.log */
+    if (strstr(reason, "Link inactive")) {
+	    statusindex = 1;
+	    saveWANStatus("Terminating connection due to lack of activity.", statusindex);
+    }
+    else {
+	    statusindex = 2;
+	    saveWANStatus(reason, statusindex);
+    }
+
     if (oldstate == STOPPED && f->flags & (OPT_PASSIVE|OPT_SILENT|DELAYED_UP)) {
 	/*
 	 * This action is not strictly according to the FSM in RFC1548,
--- pppd/pppd/utils.c
+++ pppd/pppd/utils.c
@@ -1056,3 +1056,27 @@
     }
 }
 
+/* add to wanstatus.log */
+void saveWANStatus(char *currentstatus, int statusindex)
+{
+	FILE *STATUSFILE;
+
+	/* only save with status description */
+	if (strlen(currentstatus) == 0)
+		return;
+
+#ifdef ONWL500G_SHELL
+	if ((req_unit == 0) && (STATUSFILE = fopen("/etc/linuxigd/wanstatus.log", "w")) != NULL)
+	{
+		fprintf(STATUSFILE, "StatusCode=\"%d\"\n", statusindex);
+		fprintf(STATUSFILE, "StatusReason=\"%s\"\n", currentstatus);
+		fclose(STATUSFILE);
+	}
+#else
+	if ((req_unit == 0) && (STATUSFILE = fopen("/tmp/wanstatus.log", "w")) != NULL)
+	{
+		fprintf(STATUSFILE, "%d,%s\n", statusindex, currentstatus);
+		fclose(STATUSFILE);
+	}
+#endif
+}
