diff --git pppd/pppd/auth.c pppd/pppd/auth.c
index 354f2ee..7eab985 100644
--- pppd/pppd/auth.c
+++ pppd/pppd/auth.c
@@ -332,6 +332,8 @@ option_t auth_options[] = {
       "Require CHAP authentication from peer",
       OPT_ALIAS | OPT_PRIOSUB | OPT_A2OR | MDTYPE_MD5,
       &lcp_wantoptions[0].chap_mdtype },
+    { "chap-secrets", o_string, &chapseccustom,
+      "Specify custom chap-secrets file", OPT_PRIO },
 #ifdef CHAPMS
     { "require-mschap", o_bool, &auth_required,
       "Require MS-CHAP authentication from peer",
@@ -1709,7 +1711,7 @@ have_chap_secret(char *client, char *server,
 	}
     }
 
-    filename = _PATH_CHAPFILE;
+    filename = chapseccustom ? chapseccustom : _PATH_CHAPFILE;
     f = fopen(filename, "r");
     if (f == NULL)
 	return 0;
@@ -1795,7 +1797,7 @@ get_secret(int unit, char *client, char *server,
 	    return 0;
 	}
     } else {
-	filename = _PATH_CHAPFILE;
+	filename = chapseccustom ? chapseccustom : _PATH_CHAPFILE;
 	addrs = NULL;
 	secbuf[0] = 0;
 
diff --git pppd/pppd/options.c pppd/pppd/options.c
index a4d5d6d..8829879 100644
--- pppd/pppd/options.c
+++ pppd/pppd/options.c
@@ -106,6 +106,8 @@ bool	persist = 0;		/* Reopen link after it goes down */
 char	our_name[MAXNAMELEN];	/* Our name for authentication purposes */
 bool	demand = 0;		/* do dial-on-demand */
 char	*ipparam = NULL;	/* Extra parameter for ip up/down scripts */
+char	*chapseccustom = NULL;	/* Custom chap-secrets file */
+
 int	idle_time_limit = 0;	/* Disconnect if idle for this many seconds */
 int	holdoff = 30;		/* # seconds to pause before reconnecting */
 bool	holdoff_specified;	/* true if a holdoff value has been given */
diff --git pppd/pppd/pppd.h pppd/pppd/pppd.h
index e06eca2..5b12d2c 100644
--- pppd/pppd/pppd.h
+++ pppd/pppd/pppd.h
@@ -313,6 +313,7 @@ extern char	remote_name[MAXNAMELEN]; /* Peer's name for authentication */
 extern bool	explicit_remote;/* remote_name specified with remotename opt */
 extern bool	demand;		/* Do dial-on-demand */
 extern char	*ipparam;	/* Extra parameter for ip up/down scripts */
+extern char	*chapseccustom;	/* Custom chap-secrets file */
 extern bool	cryptpap;	/* Others' PAP passwords are encrypted */
 extern int	idle_time_limit;/* Shut down link if idle for this long */
 extern int	holdoff;	/* Dead time before restarting */
