diff --git pppd/pppd/Makefile.linux pppd/pppd/Makefile.linux
index 22837c5..6d62dc4 100644
--- pppd/pppd/Makefile.linux
+++ pppd/pppd/Makefile.linux
@@ -49,7 +49,7 @@ MPPE=y
 # Uncomment the next line to include support for PPP packet filtering.
 # This requires that the libpcap library and headers be installed
 # and that the kernel driver support PPP packet filtering.
-FILTER=y
+#FILTER=y
 
 # Uncomment the next line to enable multilink PPP (enabled by default)
 # Linux distributions: Please leave multilink ENABLED in your builds
@@ -66,9 +66,9 @@ USE_TDB=y
 # pppd will not detach and will notify systemd when up.
 #SYSTEMD=y
 
-HAS_SHADOW=y
+#HAS_SHADOW=y
 #USE_PAM=y
-HAVE_INET6=y
+#HAVE_INET6=y
 
 # Enable plugins
 PLUGIN=y
@@ -85,15 +85,15 @@ USE_LIBUTIL=y
 endif
 
 # Enable EAP-TLS authentication (requires MPPE support, libssl and libcrypto)
-USE_EAPTLS=y
+#USE_EAPTLS=y
 
-MAXOCTETS=y
+#MAXOCTETS=y
 
 INCLUDE_DIRS= -I../include
 
-COMPILE_FLAGS= -DHAVE_PATHS_H -DIPX_CHANGE -DHAVE_MMAP -pipe
+COMPILE_FLAGS= -pipe #-DHAVE_PATHS_H -DIPX_CHANGE -DHAVE_MMAP
 
-CFLAGS= $(COPTS) $(COMPILE_FLAGS) $(INCLUDE_DIRS) '-DDESTDIR="@DESTDIR@"'
+CFLAGS= $(COPTS) $(COMPILE_FLAGS) $(EXTRACFLAGS) $(INCLUDE_DIRS) '-DDESTDIR="@DESTDIR@"'
 
 ifdef CHAPMS
 CFLAGS   += -DCHAPMS=1
