diff --git pppd/pppd/main.c pppd/pppd/main.c
index 71398de..90393b0 100644
--- pppd/pppd/main.c
+++ pppd/pppd/main.c
@@ -407,6 +407,13 @@ main(int argc, char *argv[])
 	fd_devnull = i;
     }
 
+    /*
+     * pppd sends signals to the whole process group, so it must always
+     * create a new one or it may kill the parent process and its siblings.
+     */
+    setsid();
+    chdir("/");
+
     /*
      * Initialize system-dependent stuff.
      */
@@ -766,8 +773,6 @@ detach(void)
 	create_linkpidfile(pid);
 	exit(0);		/* parent dies */
     }
-    setsid();
-    chdir("/");
     dup2(fd_devnull, 0);
     dup2(fd_devnull, 1);
     dup2(fd_devnull, 2);
