diff --git pppd/pppd/sys-linux.c pppd/pppd/sys-linux.c
index aeae6d2..f7d5ed9 100644
--- pppd/pppd/sys-linux.c
+++ pppd/pppd/sys-linux.c
@@ -1763,6 +1763,7 @@ int cifdefaultroute (int unit, u_int32_t ouraddr, u_int32_t gateway)
 	SIN_ADDR(rt.rt_genmask) = 0L;
     }
 
+    rt.rt_dev = ifname;
     rt.rt_flags = RTF_UP;
     if (ioctl(sock_fd, SIOCDELRT, &rt) < 0 && errno != ESRCH) {
 	if (still_ppp()) {
