--- pppd/pppd/main.c
+++ pppd/pppd/main.c
@@ -1595,10 +1595,14 @@
 	kill_my_pg(sig);
     notify(sigreceived, sig);
     if (waiting) {
+#if defined(__GNUC__) && __GNUC__ >= 5
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-result"
+#endif
 	write(sigpipe[1], &sig, sizeof(sig));
+#if defined(__GNUC__) && __GNUC__ >= 5
 #pragma GCC diagnostic pop
+#endif
     }
 }
 
@@ -1619,10 +1623,14 @@
 	kill_my_pg(sig);
     notify(sigreceived, sig);
     if (waiting) {
+#if defined(__GNUC__) && __GNUC__ >= 5
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-result"
+#endif
 	write(sigpipe[1], &sig, sizeof(sig));
+#if defined(__GNUC__) && __GNUC__ >= 5
 #pragma GCC diagnostic pop
+#endif
     }
 }
 
@@ -1636,10 +1644,14 @@
 {
     got_sigchld = 1;
     if (waiting) {
+#if defined(__GNUC__) && __GNUC__ >= 5
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-result"
+#endif
 	write(sigpipe[1], &sig, sizeof(sig));
+#if defined(__GNUC__) && __GNUC__ >= 5
 #pragma GCC diagnostic pop
+#endif
     }
 }
 
@@ -1673,10 +1685,14 @@
 {
     got_sigusr2 = 1;
     if (waiting) {
+#if defined(__GNUC__) && __GNUC__ >= 5
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-result"
+#endif
 	write(sigpipe[1], &sig, sizeof(sig));
+#if defined(__GNUC__) && __GNUC__ >= 5
 #pragma GCC diagnostic pop
+#endif
     }
 }
 
