--- wsdd2/wsd.c
+++ wsdd2/wsd.c
@@ -162,6 +162,7 @@
 	}
 }
 
+#ifndef TOMATO
 static struct {
 	const char *key, *_default;
 	char *value;
@@ -175,6 +176,21 @@
 	{ .key	= "presentationurl:",	._default = NULL},
 	{}
 };
+#else
+static struct {
+	const char *key, *_default;
+	char *value;
+} bootinfo[] = {
+	{ .key	= "vendor:",	._default = "FRESHTOMATO"},
+	{ .key	= "model:",	._default = "FreshTomato"},
+	{ .key	= "serial:",	._default = "0"},
+	{ .key	= "sku:",	._default = "FreshTomato router"},
+	{ .key	= "vendorurl:",	._default = "https://freshtomato.org"},
+	{ .key	= "modelurl:",	._default = "https://freshtomato.org"},
+	{ .key	= "presentationurl:",	._default = "http://freshtomato.org"},
+	{}
+};
+#endif /* TOMATO */
 
 int set_getresp(const char *str, const char **next)
 {
@@ -748,7 +764,11 @@
 {
 	const char resp_hdr_fmt[] =
 		"HTTP/1.1 %s\r\n"
+#ifdef TOMATO
+		"Server: FreshTomato WSD Server\r\n"
+#else
 		"Server: NETGEAR WSD Server\r\n"
+#endif
 		"Date: %s\r\n"
 		"Connection: close\r\n"
 		"Content-Type: application/soap+xml\r\n"
--- wsdd2/wsdd2.c
+++ wsdd2/wsdd2.c
@@ -581,6 +581,7 @@
 	}
 }
 
+#ifndef TOMATO
 static char *get_smbparm(const char *name, const char *_default)
 {
 #define __FUNCTION__	"get_smbparm"
@@ -616,6 +617,103 @@
 	return result;
 #undef __FUNCTION__
 }
+#else
+void trim(char *s)
+{
+	char *p = s;
+	int l = strlen(p);
+
+	if (l == 0) return;
+	while (isspace(p[l - 1])) p[--l] = 0;
+	while (* p && isspace(* p)) ++p, --l;
+
+	memmove(s, p, l + 1);
+}
+
+char *get_smbparm(const char *name, const char *_default)
+{
+#define __FUNCTION__	"get_smbparm"
+	char buf[256], *result;
+	FILE *fp;
+	char parm[64];
+	char value[256];
+	char *p, *dstp, *dstv;
+	int stage, c;
+
+	*value = '\0';
+	*parm = '\0';
+
+	if (!(fp = fopen("/etc/smb.conf", "r"))) {
+		DEBUG(0, W, __FUNCTION__ ": can't access smb.conf");
+		return strdup(_default);
+	}
+
+	while (fgets(buf, sizeof(buf), fp)) {
+		stage = 0;
+		c = 0;
+		p = buf;
+		dstp = parm;
+		dstv = value;
+
+		/* Retrieve name */
+		while (*p && c < sizeof(parm) - 1) {
+			if (stage == 0 && isspace(*p)) {
+				p++;
+				continue;
+			}
+			else
+				stage = 1;
+
+			if (*p == '=') {
+				p++; /* Skip it */
+				break;
+			}
+			*dstp++ = *p++;
+			c++;
+		}
+		*dstp = '\0';
+
+		trim(parm);
+
+		/* Is it the desired parameter? */
+		if (strcmp(parm, name))
+			continue;
+
+		/* Retrieve value */
+		stage = 0;
+		c = 0;
+
+		while (*p && c < sizeof(parm) - 1) {
+			if (stage == 0 && isspace(*p)) {
+				p++;
+				continue;
+			}
+			else
+				stage = 1;
+
+			if (*p == '\n')
+				break;
+
+			*dstv++ = *p++;
+			c++;
+		}
+		*dstv = '\0';
+
+		trim(value);
+
+		break;
+	}
+	fclose(fp);
+
+	if (!*value)
+		result = strdup(_default);
+	else
+		result = strdup(value);
+
+	return result;
+#undef __FUNCTION__
+}
+#endif /* TOMATO */
 
 static void help(const char *prog, int ec, const char *fmt, ...)
 {
--- wsdd2/wsdd.h
+++ wsdd2/wsdd.h
@@ -27,7 +27,9 @@
 #include <net/if.h> // IFNAMSIZ
 #include <arpa/inet.h> // ntohs()
 #include <netinet/in.h> // struct sockaddr_in, struct ip_mreq
+#ifndef TOMATO
 #include <linux/in.h> // struct ip_mreqn
+#endif
 #include <linux/netlink.h> // struct sockaddr_nl
 #include <time.h> // time_t, time()
 
