diff -Naur linux-6.10.3.orig/include/net/netfilter/nf_conntrack.h linux-6.10.3/include/net/netfilter/nf_conntrack.h
--- linux-6.10.3.orig/include/net/netfilter/nf_conntrack.h	2024-08-09 10:52:05.181710720 -0600
+++ linux-6.10.3/include/net/netfilter/nf_conntrack.h	2024-08-09 10:54:50.478009789 -0600
@@ -382,6 +382,11 @@
 int nf_ct_handle_fragments(struct net *net, struct sk_buff *skb,
 			   u16 zone, u8 family, u8 *proto, u16 *mru);
 
+#ifdef CONFIG_NF_CONNTRACK_DESTROY_HOOK
+void register_nf_ct_destroy_hook(void (*hook)(struct nf_conn *));
+void unregister_nf_ct_destroy_hook(void);
+#endif
+
 #define NF_CT_STAT_INC(net, count)	  __this_cpu_inc((net)->ct.stat->count)
 #define NF_CT_STAT_INC_ATOMIC(net, count) this_cpu_inc((net)->ct.stat->count)
 #define NF_CT_STAT_ADD_ATOMIC(net, count, v) this_cpu_add((net)->ct.stat->count, (v))
diff -Naur linux-6.10.3.orig/net/netfilter/Kconfig linux-6.10.3/net/netfilter/Kconfig
--- linux-6.10.3.orig/net/netfilter/Kconfig	2024-08-09 09:07:01.011027812 -0600
+++ linux-6.10.3/net/netfilter/Kconfig	2024-08-09 10:54:50.478009789 -0600
@@ -79,11 +79,15 @@
 	  If this option is enabled, the kernel will include support
 	  for passive OS fingerprint via NFNETLINK.
 
+config NF_CONNTRACK_DESTROY_HOOK
+	bool
+
 config NF_CONNTRACK
 	tristate "Netfilter connection tracking support"
 	default m if NETFILTER_ADVANCED=n
 	select NF_DEFRAG_IPV4
 	select NF_DEFRAG_IPV6 if IPV6 != n
+	select NF_CONNTRACK_DESTROY_HOOK
 	help
 	  Connection tracking keeps a record of what packets have passed
 	  through your machine, in order to figure out how they are related
diff -Naur linux-6.10.3.orig/net/netfilter/nf_conntrack_core.c linux-6.10.3/net/netfilter/nf_conntrack_core.c
--- linux-6.10.3.orig/net/netfilter/nf_conntrack_core.c	2024-08-09 09:07:01.011027812 -0600
+++ linux-6.10.3/net/netfilter/nf_conntrack_core.c	2024-08-09 10:54:50.478009789 -0600
@@ -571,9 +571,30 @@
 #endif
 }
 
+#ifdef CONFIG_NF_CONNTRACK_DESTROY_HOOK
+
+static void (*nf_ct_destroy_hook)(struct nf_conn *) __rcu __read_mostly = NULL;
+
+void register_nf_ct_destroy_hook(void (*hook)(struct nf_conn *))
+{
+	rcu_assign_pointer(nf_ct_destroy_hook, hook);
+}
+EXPORT_SYMBOL(register_nf_ct_destroy_hook);
+
+void unregister_nf_ct_destroy_hook(void)
+{
+	rcu_assign_pointer(nf_ct_destroy_hook, NULL);
+}
+
+EXPORT_SYMBOL(unregister_nf_ct_destroy_hook);
+#endif
+
 void nf_ct_destroy(struct nf_conntrack *nfct)
 {
 	struct nf_conn *ct = (struct nf_conn *)nfct;
+#ifdef CONFIG_NF_CONNTRACK_DESTROY_HOOK
+	void (*hook)(struct nf_conn *);
+#endif
 
 	WARN_ON(refcount_read(&nfct->use) != 0);
 
@@ -582,6 +603,12 @@
 		return;
 	}
 
+#ifdef CONFIG_NF_CONNTRACK_DESTROY_HOOK
+	hook = rcu_dereference(nf_ct_destroy_hook);
+	if (hook)
+		hook(ct);
+#endif
+
 	if (unlikely(nf_ct_protonum(ct) == IPPROTO_GRE))
 		destroy_gre_conntrack(ct);
 
